server/bin/pbench-sync-satellite.sh

```diff
@@ -250,10 +253,26 @@ for host in $hosts ;do
         for x in $state_list; do
             echo "$host/TO-SYNC/$x" >> ${state_change_log}
         done
+        rm $logdir/$host/ok-checks.log
+    fi
+
+    if [[ ! -s $logdir/$host/fail-checks.log ]] ; then
+        rm $logdir/$host/fail-checks.log
+    fi
+
+    if [[ -z $(grep -E -v -m 1 'OK|FAILED' $logdir/$host/md5-checks.log) ]] ; then
+        rm $logdir/$host/md5-checks.log
     fi
```

Comment:
```
    if [[ ! -s $logdir/$host/fail-checks.log ]] ; then
        rm $logdir/$host/fail-checks.log
        # Look for all lines in md5-checks.log that DO NOT match "OK", where
        # "grep" will return 0 when it does find such lines, and non-zero when
        # it does not find such lines.
        grep -q -F -v 'OK' $logdir/$host/md5-checks.log
        if [[ $? -ne 0 ]]; then
            # We can safely remove md5-checks.log because grep DID NOT find
            # any lines other than those that contained "OK".
            rm $logdir/$host/md5-checks.log
        fi
    fi
```